src/tools/processData.js

```diff
@@ -0,0 +1,301 @@
+    );
+  }
+
+  await Promise.all(promises);
+  // Retrieve transformed users
+  transformedUsers = await User.findAll({
+    attributes: ['id', 'email'],
+  }).map((u) => u.dataValues);
+};
+
+export const hideGranteesGrants = async (granteesGrants) => {
+  realGrants = await Grant.findAll({
+    attributes: ['id', 'granteeId', 'number'],
+  }).map((g) => g.dataValues);
+
+  const granteesArray = granteesGrants ? granteesGrants.split('\n').map((el) => el.split('|')[0].trim()) : null;
+  const grantsArray = (granteesArray && granteesArray.length > 1) ? granteesGrants.split('\n').map((el) => el.split('|')[1].trim()) : null;
+  const granteeWhere = granteesArray ? { name: { [Op.like]: { [Op.any]: granteesArray } } } : {};
+  const grantWhere = grantsArray ? { number: { [Op.like]: { [Op.any]: grantsArray } } } : {};
+  const grantees = await Grantee.findAll({
+    where: granteeWhere,
+  });
+
+  const promises = [];
+
+  // loop through the found reports
+  for await (const grantee of grantees) {
+    promises.push(
+      grantee.update({
+        name: faker.company.companyName(),
+      }),
+    );
+  }
+  const grants = await Grant.findAll({
+    where: grantWhere,
+  });
+
+  for await (const grant of grants) {
+    const trailingNumber = grant.id;
+    const newGrantNumber = `0${faker.datatype.number({ min: 1, max: 9 })}${faker.animal.type()}0${trailingNumber}`;
+
+    promises.push(
+      grant.update({
+        number: newGrantNumber,
+      }),
+    );
+  }
+  await Promise.all(promises);
+
+  // Retrieve transformed grantees
+  transformedGrantees = await Grantee.findAll({
+    attributes: ['id', 'name'],
+    where: { id: grantees.map((g) => g.id) },
+  }).map((g) => g.dataValues);
+
+  // Retrieve transformed grants
+  transformedGrants = await Grant.findAll({
+    attributes: ['id', 'number'],
+    where: { id: grants.map((g) => g.id) },
+  }).map((g) => g.dataValues);
+};
+const processData = async (mockReport) => {
+  const activityReportId = mockReport ? mockReport.id : null;
+  const where = activityReportId ? { id: activityReportId } : {};
+  const filesWhere = activityReportId ? { activityReportId } : {};
+  const userIds = mockReport ? [3000, 3001, 3002, 3003] : null;
+
+  const granteesGrants = mockReport ? mockReport.imported.granteeName : null;
+  const reports = await ActivityReport.unscoped().findAll({
+    where,
+  });
+
+  const files = await File.findAll({
+    where: filesWhere,
+  });
+
+  const promises = [];
+
+  // Hide users
+  await hideUsers(userIds);
+  // Hide grantees and grants
+  await hideGranteesGrants(granteesGrants);
+
+  // loop through the found reports
+  for await (const report of reports) {
+    const { imported } = report;
+
+    promises.push(
+      report.update({
+        managerNotes: await processHtml(report.managerNotes),
+        additionalNotes: await processHtml(report.additionalNotes),
+        context: await processHtml(report.context),
+      }),
+    );
+    if (imported) {
```

Comment:
Seems like a lot of complexity of this script centers around the imported reports. May not be worth it, but can we make the script simpler by just deleting all legacy reports and adding some number of completely made up legacy reports

Reply:
Interesting idea. I am thinking that since we have the logic already, it could serve us by providing a more realistic view of production for data mining, especially the goals, user, and grantee/grant statistics.